### ISSUE TITLE
Error when I deploy using "sls resources deploy"

### ISSUE THREAD

**Issue by varunvairavan** [serverless/serverless#809]
```
ServerlessError: ServerlessError: An error occurred while provisioning your cloudformation: The following resource(s) failed to create: [IamRoleLambda]. . Delete requested by user.
    at new ServerlessError (/usr/local/lib/node_modules/serverless/lib/ServerlessError.js:17:11)
    at /usr/local/lib/node_modules/serverless/lib/actions/ResourcesDeploy.js:265:19
From previous event:
    at /usr/local/lib/node_modules/serverless/lib/actions/ResourcesDeploy.js:260:17
From previous event:
    at ResourcesDeploy._deployResources (/usr/local/lib/node_modules/serverless/lib/actions/ResourcesDeploy.js:172:10)
From previous event:
    at ResourcesDeploy.resourcesDeploy (/usr/local/lib/node_modules/serverless/lib/actions/ResourcesDeploy.js:79:10)
From previous event:
    at /usr/local/lib/node_modules/serverless/lib/Serverless.js:157:29
    at Array.reduce (native)
    at /usr/local/lib/node_modules/serverless/lib/Serverless.js:156:30
    at processImmediate [as _immediateCallback] (timers.js:368:17)
From previous event:
    at Serverless._execute (/usr/local/lib/node_modules/serverless/lib/Serverless.js:154:10)
    at Serverless.actions.(anonymous function) (/usr/local/lib/node_modules/serverless/lib/Serverless.js:382:20)
    at Serverless.command (/usr/local/lib/node_modules/serverless/lib/Serverless.js:351:38)
    at Object.<anonymous> (/usr/local/lib/node_modules/serverless/bin/serverless:16:12)
    at Module._compile (module.js:435:26)
    at Object.Module._extensions..js (module.js:442:10)
    at Module.load (module.js:356:32)
    at Function.Module._load (module.js:311:12)
    at Function.Module.runMain (module.js:467:10)
    at startup (node.js:134:18)
    at node.js:961:3
```
I deleted the region by mistake using `sls region remove -s dev -r us-east-1`, I re-created the region using `sls region create`, but when I deploy I get this error.

It seems similar to [This](https://github.com/serverless/serverless-graphql-blog/issues/5) issue, But I do not want to move to a different region. Thoughts?

**Comment by adrianmurillo** [comment]
I got the same error, after deleting some DynamoDB resources using the CLI. Even if I moved to another region, I still got it.

I searched high and low for an answer, to no avail. The few answers I found, didn't work (like that one about changing regions).

What I did to avoid the error, is remove the DynamoDB declarations. This didn't fix the problem, and forced me to create the resources manually, but at least I can keep working.

**Comment by ajagnanan** [comment]
Did you guys look at Cloudformation to see what it is erroring out on? If you make certain changes, it requires you to delete the stack manually and re-create it with `sls resources deploy`

**Comment by varunvairavan** [comment]
I uploaded the `s-resources-cf-dev-useast1.json` to Cloudformation and it successfully created the stack. No errors. When I run `sls resoures deploy` with the stack already created, it still throws the same error. I also tried deleting all the IAM policies, S3 buckets, and Cloudformation template and tried running `sls resources deploy` with the same result. Also tried removing and creating the region again with the same result.

The exact steps that put me here,

I ran `sls resources remove`. Serverless said 

`Successfully removed "dev" resources from "us-east-1"`

But then when I ran `sls resources deploy` it hit an error, It was unable to delete the IAM policies, and it would not allow me to run deploy or delete again.

`ServerlessError: Stack:arn:aws:cloudformation:us-east-1:abcd is in DELETE_FAILED state and can not be updated.`

I checked Cloudformation and it said it was unable to delete the IAM roles. So I deleted the IAM roles myself. Then deleted the S3 buckets and the Cloudformation templates. I also ran `sls resources delete` again and it said it was successful. 

When I tried to deploy again it errored out saying 

`The following resource(s) failed to create: [IamRoleLambda]. . Delete requested by user.`

**Comment by adrianmurillo** [comment]
Yes. Same here. I really think the `resources remove` function is either faulty or unclear on its results.

**Comment by varunvairavan** [comment]
SOLVED. The user I was using to deploy did not have permissions to make changes to IAM roles. 
    `[
        "iam:GetRole",
        "iam:CreateRole",
        "iam:PutRolePolicy",
        "iam:DeleteRolePolicy"
    ]`

How I got into this mess
1) `sls resources remove`. The command looked like it succeeded by it had failed because it could not delete the IAM role. Serverless did not print that error. I then deleted the IAM role and the template myself using the AWS console.

2) I deleted and re-created the region with the `sls` command. I do not think this was the cause of any issue.

3) I tried deploying, but it kept failing with the error `The following resource(s) failed to create: [IamRoleLambda]. . Delete requested by user.` This error was again because of permissions. I failed to see that because of the misleading "Delete requested by user", which made me believe it had something to do with the previous issues.

4) I uploaded the Cloudformation template directly using the console and the creation was successful. But my local serverless code was not linked to this stack. I was also unable to deploy functions anymore, failed with the error `Missing required key 'Role' in params`

How I fixed it
1) I updated `s-variables-dev-useast1.json` with the Cloudformation arn that got created when I uploaded the template directly and the IAM role that got created. `resourcesStackName` and `iamRoleArnLambda`. I was now able to deploy the functions.
2) I gave the `iam` roles mentioned at the top to the serverless user. Deploy also started working.

**Comment by sielay** [comment]
Thanks for tip. That can be any error. I had situation that deletion process didn't took one S3 bucket down and wasn't able to recreate stack.

**Comment by calam1** [comment]
Ok, I was having similar problems with errors like:

The following resource(s) failed to create: [IamRoleLambda]. . Delete requested by user.

I took a look at the cloudformation logs, under the "deleted" filter.  I saw that the process was failing due to the resource existing, but it was under another stack.  I was experimenting earlier and manually running cloudformation templates via the console.  

So what I did in the command line, you must have the aws-cli installed, was aws cloudformation describe-stacks and to my surprise I found a couple of outstanding stacks with created resources associated with it.  I deleted those stacks via the command line, aws cloudformation delete-stack --stack-name xyz

i ran sls project remove, I went into my console and made sure everything was deleted, i.e. dynamodb tables, topics, as you know the remove does not delete dynamodb tables, etc;  make sure you are starting from a clean slate.

Then I ran sls project init, picked the defaults as usual, YMMV, and it started working again.

Hope this helps people, this was driving me nuts for a day or two.
